src/lib.rs

```diff
@@ -242,7 +242,125 @@ pub trait PolynomialCommitment<F: PrimeField, P: Polynomial<F>, S: Cryptographic
     where
         Self::Commitment: 'a;
 
-    /// batch_check but with individual challenges
+    /// Open several polynomials at one or more points each (possibly different
+    /// for each polynomial). Each entry in the in the query set of points
+    /// contains the label of the polynomial which should be queried at that
+    /// point.
+    ///
+    /// Behaviour is undefined if `query_set` contains the entries with the
+    /// same point label but different actual points.
+    ///
+    /// The opening challenges are independent for each batch of polynomials.
+    ///
+    /// The default implementation achieves this by rearranging the queries in
+    /// order to gather (i.e. batch) all polynomials that should be queried at
+    /// the same point, then opening their commitments simultaneously with a
+    /// single call to `open` (per point)
```

Comment:
@Antonio95 as you pointed out offline, I think this indeed belongs to the inline comment of the default implementation rather than on the trait.

Reply:
Agreed. Done! thanks